features/FEATURE_BLE/source/generic/GenericGap.cpp

```diff
@@ -1555,5 +1641,970 @@ void GenericGap::set_connection_event_handler(pal::ConnectionEventMonitor::Event
+    uint8_t number_of_completed_extended_advertising_events
+)
+{
+    _active_sets.clear(advertising_handle);
+
+    if (!_eventHandler) {
+        return;
+    }
+
+    _eventHandler->onAdvertisingEnd(
+        AdvertisingEndEvent_t(
+            advertising_handle,
+            connection_handle,
+            number_of_completed_extended_advertising_events,
+            status == pal::hci_error_code_t::SUCCESS
+        )
+    );
+}
+
+void GenericGap::on_scan_request_received(
+    advertising_handle_t advertising_handle,
+    connection_peer_address_type_t scanner_address_type,
+    const ble::address_t &address
+)
+{
+    if (!_eventHandler) {
+        return;
+    }
+
+    _eventHandler->onScanRequest(
+        ScanRequestEvent_t(
+            advertising_handle,
+            (ble::peer_address_type_t::type) scanner_address_type.value(),
+            address
+        )
+    );
+}
+
+ble_error_t GenericGap::setScanParameters(const ScanParameters &params)
+{
+    use_non_deprecated_scan_api();
+
+    if (is_extended_advertising_available()) {
+        bool active_scanning[] = {
+            params.get_1m_configuration().active_scanning,
+            params.get_coded_configuration().active_scanning
+        };
+
+        uint16_t scan_interval[] = {
+            params.get_1m_configuration().interval.value(),
+            params.get_coded_configuration().interval.value()
+        };
+
+        uint16_t scan_window[] = {
+            params.get_1m_configuration().window.value(),
+            params.get_coded_configuration().window.value()
+        };
+
+        return _pal_gap.set_extended_scan_parameters(
+            (pal::own_address_type_t::type) params.get_own_address_type(),
+            (pal::scanning_filter_policy_t::type) params.get_scanning_filter_policy(),
+            params.get_scanning_phys(),
+            active_scanning,
+            scan_interval,
+            scan_window
+        );
+    } else {
+        if (params.get_scanning_phys().get_coded()) {
+            return BLE_ERROR_INVALID_PARAM;
+        }
+
+        ScanParameters::phy_configuration_t legacy_configuration =
+            params.get_1m_configuration();
+
+        return _pal_gap.set_scan_parameters(
+            legacy_configuration.active_scanning,
+            legacy_configuration.interval.value(),
+            legacy_configuration.window.value(),
+            (pal::own_address_type_t::type) params.get_own_address_type(),
+            (pal::scanning_filter_policy_t::type) params.get_scanning_filter_policy()
+        );
+    }
+}
+
+ble_error_t GenericGap::startScan(
+    scanning_filter_duplicates_t filtering,
+    UnitScanDuration_t duration,
+    UnitScanPeriod_t period
+)
+{
+    use_non_deprecated_scan_api();
+
+    if(_privacy_enabled && _central_privacy_configuration.use_non_resolvable_random_address)
+    {
+        set_random_address_rotation(true);
+    }
+
+    if (is_extended_advertising_available()) {
+        return _pal_gap.extended_scan_enable(
```

Comment:
Parameters are rejected on legacy controllers if they include phy parameters: 

https://github.com/ARMmbed/mbed-os/blob/3a2fb280f07770d6f5fa17f0465c716203665d22/features/FEATURE_BLE/source/generic/GenericGap.cpp#L2383-L2385

Reply:
they are rejected in the enhanced call (and not really since the feature has already been checked here to be present) but not in the normal scan enable call below